icu4c/source/i18n/rbt_pars.cpp

```diff
@@ -976,9 +976,12 @@ void TransliteratorParser::parseRules(const UnicodeString& rule,
             if (!parsingIDs) {
                 if (curData != NULL) {
                     if (direction == UTRANS_FORWARD)
-                        dataVector.addElementX(curData, status);
+                        dataVector.addElement(curData, status);
                     else
                         dataVector.insertElementAt(curData, 0, status);
+                    if (U_FAILURE(status)) {
+                        delete curData;
```

Comment:
I wonder if maybe adding a `U_ASSERT(!dataVector.hasDeleter());` here?

If I understand the new approach:
- `addElement` doesn't delete the object.
- `insertElementAt` does delete (if there is a deleter).

(in case somehow the `dataVector` does end up with a deleter, to prevent a double-delete ?)